os/arch/arm/src/amebad/amebad_timer_lowerhalf.c

```diff
@@ -0,0 +1,348 @@
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_gpt_start(struct timer_lowerhalf_s *lower)
+{
+	struct amebad_gpt_lowerhalf_s *priv = (struct amebad_gpt_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	if (!priv->started) {
+		if (priv->callback != NULL) {
+			gtimer_init(&priv->obj, priv->obj.timer_id);
+		}
+		gtimer_start_periodical(&priv->obj, priv->gpt_timeout, (void *)amebad_timout_handler, priv);
+		priv->started = true;
+		gtimer_reload(&priv->obj, 1000000);
+		tmrvdbg("Timer %d is started, callback %s\n", priv->obj.timer_id, priv->callback == NULL ? "none" : "set");
+		return OK;
+	}
+
+	/* Return EBUSY to indicate that the timer was already running */
+	tmrdbg("Error!! Timer %d was already running\n", priv->obj.timer_id);
+	return -EBUSY;
+}
+
+/****************************************************************************
+ * Name: amebad_gpt_stop
+ *
+ * Description:
+ *   Stop the timer
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_gpt_stop(struct timer_lowerhalf_s *lower)
+{
+	struct amebad_gpt_lowerhalf_s *priv = (struct amebad_gpt_lowerhalf_s *)lower;
+
+	if (priv->started) {
+		gtimer_deinit(&priv->obj);
+		priv->started = false;
+
+		tmrvdbg("Timer %d is stopped\n", priv->obj.timer_id);
+		return OK;
+	}
+
+	/* Return ENODEV to indicate that the timer was not running */
+	tmrdbg("Error!! Timer %d is not running\n", priv->obj.timer_id);
+	return -ENODEV;
+}
+
+/****************************************************************************
+ * Name: amebad_gpt_settimeout
+ *
+ * Description:
+ *   Set a new timeout value (and reset the timer)
+ *
+ * Input Parameters:
+ *   lower   - A pointer the publicly visible representation of the
+ *             "lower-half" driver state structure.
+ *   timeout - The new timeout value in microseconds.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_gpt_settimeout(struct timer_lowerhalf_s *lower, uint32_t timeout)
+{
+	struct amebad_gpt_lowerhalf_s *priv = (struct amebad_gpt_lowerhalf_s *)lower;
+
+	if (priv->started) {
+		return -EPERM;
+	}
+
+	gtimer_reload(&priv->obj, timeout);
+	priv->gpt_timeout = timeout;
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: amebad_gpt_setcallback
+ *
+ * Description:
+ *   Call this user provided timeout callback.
```

Comment:
This function sets the user provided callback, not call that.

Reply:
Noted, for that, I just followed the common description used by other platforms.  
I will change the description to 'Set the user provided timeout callback.' Thanks!